File: pset1/mario.c
#include<stdio.h>
#include<cs50.h>

int main(void)
{
    int height = 0 ;
    int spaces = 0 ;
    int hashes = 0 ;
  /*
  
  asking for the height .
  and check if its a negtive number or greater than 23.
  
  */
    do
    {
        printf("Please give me the Height of the half pyramid,must be Postive Number And Not bigger than 23 \n");
        height = GetInt();
    }
    while (height < 0 || height > 23);
  
  
    spaces = height - 1 ; 
    hashes = 2 ;
    // loop for the height of the pyramid.
    for (int i = 0;i<height;i++)
    {
        //loop for number of spaces of each row.
        for(int j = 0 ; j < spaces ; j++)
            printf(" ");
        //loop for number of hashes of each row.
        for(int j = 0 ; j < hashes ; j++)
            printf("#");
            
        
        printf("\n");
        spaces--;
        hashes++;
    }
}

File: pset6/mario.py
import sys

hight = 0
spaces = 0
hashes = 0

while True :
    hight = int(input("Please give me the Height of the half pyramid,must be Postive Number And Not bigger than 23\n"))
    if hight > 0 and hight < 23 :
        break
    
spaces = hight - 1 
hashes = 2

for i in range(hight) :
    for j in range(spaces) :
        print(" ",end="")
    for x in range(hashes) :
        print("#",end="")
        
    print("\n",end="")
    spaces-=1
    hashes+=1
    